apps/qubit/modules/default/actions/bs2DeprecationMessageComponent.class.php

```diff
@@ -26,13 +26,14 @@ class DefaultBS2DeprecationMessageComponent extends sfComponent
 {
     public function execute($request)
     {
-        if (!$this->context->user->isAuthenticated() || sfConfig::get('app_b5_theme', false)) {
-            return sfView::NONE;
-        }
+        $hasTranslateOrEditAccess = QubitAcl::check($this->resource, 'update')
```

Comment:
I'm not sure if we should check against `$this->resource`. This component may be called from all pages, so I'm not sure if a resource will exist in all cases.

Reply:
Ah, you're right! I'm updating it to simply test against `QubitInformationObject` instead, which is something I'd seen in a couple other places [such as in the sfIsdiahPlugin](https://github.com/artefactual/atom/blob/c088704e63506d6bca5d9ea526e04fe59f12c5f9/plugins/arDominionB5Plugin/modules/sfIsdiahPlugin/templates/indexSuccess.php#L280). It seems to work, but let me know if you have a better suggestion!

Reply:
Removing this breaks the check for admin accounts, so I'm adding the admin check in.